libs/s25main/ai/aijh/AIPlayerJH.cpp

```diff
@@ -344,9 +344,8 @@ void AIPlayerJH::PlanNewBuildings(const unsigned gf)
         DistributeGoodsByBlocking(GoodType::Boards, 30);
         DistributeGoodsByBlocking(GoodType::Stones, 50);
         // go to the picked random warehouse and try to build around it
-        int randomStore = rand() % (storehouses.size());
         auto it = storehouses.begin();
-        std::advance(it, randomStore);
+        std::advance(it, AI::randomIndex(storehouses));
```

Comment:
If you want you can replace this using (a wrapper of) `helpers::getRandomElement`. But can be done later.